File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseFindSystemTimeZoneById.vb
Imports TimeZoneConverter

Public Class Program
    Public Sub Compliant()
        TZConvert.IanaToWindows("Asia/Tokyo")
        TZConvert.WindowsToIana("Asia/Tokyo")
        Dim resolvedTimeZone As String
        TZConvert.TryIanaToWindows("Asia/Tokyo", resolvedTimeZone)
        TZConvert.TryWindowsToIana("Asia/Tokyo", resolvedTimeZone)

        TZConvert.IanaToRails("Asia/Tokyo")
        TZConvert.TryRailsToIana("Asia/Tokyo", resolvedTimeZone)
    End Sub
End Class

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/UseFindSystemTimeZoneById.cs
using System;
using TimeZoneConverter;

public class Program
{
    public void Noncompliant()
    {
        TZConvert.IanaToWindows("Asia/Tokyo"); // Noncompliant {{Use "TimeZoneInfo.FindSystemTimeZoneById" instead of "TZConvert.IanaToWindows"}}
//                ^^^^^^^^^^^^^
        TZConvert.WindowsToIana("Asia/Tokyo"); // Noncompliant
        string resolvedTimeZone;
        TZConvert.TryIanaToWindows("Asia/Tokyo", out resolvedTimeZone); // Noncompliant
        TZConvert.TryWindowsToIana("Asia/Tokyo", out resolvedTimeZone); // Noncompliant
    }

    public void Compliant()
    {
        TZConvert.IanaToRails("Asia/Tokyo");
        string resolvedTimeZone;
        TZConvert.TryRailsToIana("Asia/Tokyo", out resolvedTimeZone);
    }
}
